src/Sentry/Internal/RedactedHeaders.cs

```diff
@@ -0,0 +1,51 @@
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Internal;
+
+internal class RedactedHeaders : IDictionary<string, string>
+{
+    private readonly Dictionary<string, string> _inner = new(StringComparer.OrdinalIgnoreCase);
+
+    private static string Redact(string key, string value) =>
+        string.Equals(key, "Authorization", StringComparison.OrdinalIgnoreCase) ? "[Filtered]" : value;
```

Comment:
We could be a bit more aggressive there. The [Python SDK redacts a bunch of headers](https://github.com/getsentry/sentry-python/blob/b16fa5ffbad39843ebd2e9bc4ea6e91c0c9aa192/sentry_sdk/integrations/_wsgi_common.py#L28-L35) (not just `Authorization`):